calculator/src/main/java/com/market/banica/calculator/exception/GlobalExceptionHandler.java

```diff
@@ -20,17 +20,20 @@
  */
 
 @ControllerAdvice
-public class GlobalExceptionHandler {
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GlobalExceptionHandler.class);
 
-    @ExceptionHandler({FeatureNotSupportedException.class})
-    public ResponseEntity<Object> notSupportedFeatureHandling(FeatureNotSupportedException exception) {
-        return new ResponseEntity<>(Collections.singletonMap("error", exception.getMessage()), HttpStatus.valueOf(501));
+    @ExceptionHandler(IllegalArgumentException.class)
+    protected ResponseEntity<Object> handleIllegalArgument(
+            RuntimeException runtimeException, WebRequest request) {
+        return handleExceptionInternal(runtimeException, runtimeException.getMessage(),
+                new HttpHeaders(), HttpStatus.BAD_REQUEST, request);
     }
 
     @ExceptionHandler({BadResponseException.class})
-    public void badAuroraResponse(BadResponseException exception){
-        LOGGER.warn("Exception caught: {}" ,exception.getMessage());
+    public  ResponseEntity<Object> badAuroraResponse( RuntimeException runtimeException, WebRequest request){
```

Comment:
it is a good idea to discuss with the team and choose a common formatter for all the classes - I see that there is a double space after public and a space in parameters defining. This would be good because if in future someone decides to change something in the class and uses a different 'code formatter' it will look like he is changing this method and there are going to be unnecessary big pull requests. I could provide the formatter that we are using in our team, to be common for you all, not creating a custom one

Reply:
Done